code/modules/surgery/organs/liver.dm

```diff
@@ -107,6 +107,8 @@
 
 	if(filterToxins && !HAS_TRAIT(liver_owner, TRAIT_TOXINLOVER))
 		for(var/datum/reagent/toxin/toxin in cached_reagents)
+			if(status != toxin.affected_organtype) //this particular toxin does not affect this type of organ
```

Comment:
Just want to note for other people that it's fine since both can only contain one flag